src/hb-aat-layout-trak-table.hh

```diff
@@ -124,8 +157,28 @@ struct trak
   inline bool sanitize (hb_sanitize_context_t *c) const
   {
     TRACE_SANITIZE (this);
-    /* TODO */
-    return_trace (c->check_struct (this));
+
+    if (!(c->check_struct (this)))
+      return_trace (false);
+
+    if ((format != 0) || (reserved != 0))
+      return_trace (false);
+
+    if (horizData)
+    {
+      const TrackData &trackData = this+horizData;
```

Comment:
The `&` here is important. Failing to put it there generates memory-wrong code when the structures reference other structs. I tried to make it illegal to assign these structs, but couldn't get it working.

Reply:
Public link about design of our structs? Not yet. I'm slowly trying to work on one.

Reply:
I guess I no longer need that documentation, its too late for me :)

About what was that attempt on making struct uncopyable, I know it starts with making copy method private but I like to know what stopped the move and learn from that.

Reply:
I can make it by creating a new empty type to be derived from by all font structs. Eg.:
```
struct FontStruct 
{ 
  private: 
  FontStruct (const FontStruct &); 
  FontStruct & operator = (const FontStruct &); 
}; 
```
I don't know What else we can do with that. Also not sure if the DEFINE_* should check that we inherit from this. Note that for basic types (HBINT*, Tag, GlyphID, ...) we don't want to enforce this.  So maybe we can have two different types, one for ValueTypes and one for others.  I like the idea. Can you open an issue for it?

If we make the base type a template that takes the inherited type (https://en.wikipedia.org/wiki/Curiously_recurring_template_pattern) we might be able to do more useful things as well.

Reply:
#843